File: algorithm/typescript/BOJ10815.ts
// problem : https://www.acmicpc.net/problem/10815

import fs = require("fs");

const input: string[] = fs.readFileSync("/dev/stdin").toString().split("\n");
const result: number[] = [];
const N: number = Number(input[0]);
const nCard: number[] = input[1].split(" ").map(Number);
const M: number = Number(input[2]);
const mCard: number[] = input[3].split(" ").map(Number);
const map = new Map();

nCard.forEach((number) => {
  map.set(number, 1);
});

mCard.forEach((number) => {
  const value = map.get(number) == 1 ? 1 : 0;
  result.push(value);
});

console.log(result.join(" "));

File: algorithm/python/BOJ10815.py
# problem : https://www.acmicpc.net/problem/10815

import sys

N = int(sys.stdin.readline())
l1 = list(map(int, sys.stdin.readline().split()))
d1 = dict()
for value in l1:
    d1[value] = 1

M = int(sys.stdin.readline())
l2 = list(map(int, sys.stdin.readline().split()))
for key in l2:
    if d1.get(key) != None:
        print(1, end=" ")
    else:
        print(0, end=" ")
